### PR TITLE
Make inbm-lib type checking stricter with mypy

### PR DESCRIPTION
**Pull request by gblewis1**
- [x] autopep8
- [x] integration test
- no changelog needed

### CODE REVIEW

**Comment by gblewis1** [comment]
@nmgaston please review

File: inbm-lib/inbm_lib/dbs_parser.py
Comment (nmgaston) [inline inbm-lib/inbm_lib/dbs_parser.py:29]: These will be removed with my PR.  Created a dataclass instead.  Could we remove this change to save the merge conflict?

Comment (gblewis1) [reply inbm-lib/inbm_lib/dbs_parser.py:29]: @nmgaston I can wait until your PR is merged in and then sort it out here so once this PR is in, all of inbm-lib will pass the strict checks.

**Comment by gblewis1** [comment]
@nmgaston please re-review now that your PR is in (I've done the merge here)

**Review by nmgaston — APPROVED** [review]

**Pull request merged**
